### ISSUE TITLE
Mod Support?

### ISSUE THREAD

**Issue by BinaryConstruct** [TEdit/Terraria-Map-Editor#1075]
This might be a big project, but here goes:

High Level Goals:

- [ ]  Native save and load of modded world files. 
- [ ]  Rendering of modded tiles??? 
- [ ]  Rendering of modded NPCs
- [ ]  Editor for modded world flags.
- [ ]  100% compatibility with unmodded Terraria

Known Mod Loaders for planned support (and reference code/documentation)

- tModLoader https://github.com/blushiemagic/tModLoader/blob/5ca770ccffc07a817206ddc05964a17cad091550/patches/tModLoader/Terraria.ModLoader.IO/WorldIO.cs

**Comment by BinaryConstruct** [comment]
[GitMate.io](https://gitmate.io) thinks possibly related issues are https://github.com/TEdit/Terraria-Map-Editor/issues/721 (Mac support?), https://github.com/TEdit/Terraria-Map-Editor/issues/158 (possible support for cmod?), https://github.com/TEdit/Terraria-Map-Editor/issues/882 (tModLoader support), https://github.com/TEdit/Terraria-Map-Editor/issues/355 (Support for Terraria 1.2), and https://github.com/TEdit/Terraria-Map-Editor/issues/718 (Linux support).

**Comment by Jofairden** [comment]
Hi, TML dev here. For a start, it would be nice if it just 'supported' modded things. I guess this would namely include ModItem, ModTile, ModNPC.. etc. For actual mod data, you would have to learn a bit how it is setup if you haven't. What you could do is ask players for their TML files (.tplr, .twld) which contains mod data. You can open these up in [NBTExplorer](https://github.com/jaquadro/NBTExplorer) and take a look at the contents. (the mod data is stored as a Node-Binary-Tree) I think just supporting to see (read: not crash or malfunction) modded stuff would be a great start, actually editing might be another thing you could leave on the table for now. I am actually not sure how easy or hard it would be to render things, as the files are obviously not part of the main files. Probably one way would be to have users load the .tmod files during use, because they hold the resources.

**Comment by BinaryConstruct** [comment]
Thanks for the input. Are the `.twld` files used in conjunction with the regular `.wld` files or are they stand alone?

**Comment by coachl4d2** [comment]
Is this still planning to be looked at as a feature?

**Comment by BinaryConstruct** [comment]
Yes, however there are some major technical hurdles, such as each individual mod can change the way the world is loaded.

**Comment by choskeli** [comment]
Could a first pass of this feature just fix the fact that tedit corrupts calamity chests (they cannot be opened)? Even if we have to stick to editing vanilla tiles that's still a huge improvement over no map editor at all for moded maps
